packages/angular_devkit/build_angular/src/karma/index.ts

```diff
@@ -95,6 +111,33 @@ export function execute(
         }
       }
 
+      // prepend special webpack loader that will transform test.ts
+      if (webpackConfig && webpackConfig.module && options.spec) {
+        const sourceRoot = workspace.getProject(projectName).sourceRoot || '';
+        const files = findSpecs(options.spec, workspace.root, sourceRoot);
+        // early exit, no reason to start karma
+        if (!files.length) {
+          subscriber.error(`Specified pattern: "${options.spec}" did not match any spec files`);
+
+          return;
+        }
+        // local unit test requires .ts extension
+        // without one specs will fail with unclear error
+        // release build does not need any extension
+        const ext = __filename.endsWith('.ts') ? '.ts' : '';
+
+        webpackConfig.module.rules.unshift({
+          test: /test\.ts$/,
+          use: {
+            loader: resolve(__dirname, '../angular-cli-files/plugins/single-test-transform' + ext),
```

Comment:
You can do the same thing as below to get around this problem:
https://github.com/angular/angular-cli/blob/5df02a3de5257162c7b8a5e2c65d388debcd51f0/packages/angular_devkit/build_angular/src/angular-cli-files/plugins/webpack.ts#L22

It's also a bit friendlier as it exposes it (even though it's not public API).

Reply:
Will test and see if this works in CI and local environment

Reply:
Seems to work during development and when it is build and installed